### PR TITLE
Add update_extent_srid to update srid for extent geom

### PR DESCRIPTION
**Pull request by htran84**
- add test
_ add update_extent_srid to update srid for extent geom

### CODE REVIEW

File: tethysext/atcore/models/app_users/spatial_resource.py [hunk 1]
```diff
@@ -78,3 +78,11 @@ def get_extent(self, extent_type: str = 'dict'):
         if extent_type == 'dict':
             extent = json.loads(extent)
         return extent
+
+    def update_extent_srid(self, srid):
+        session = inspect(self).session
+        object_to_convert = session.query(Resource).filter(Resource.id == self.id).first()
```
Comment (swainn) [inline tethysext/atcore/models/app_users/spatial_resource.py:84]: I'm pretty sure `object_to_convert` is `self`, no? If you need a new copy, use the `copy` module.

Comment (swainn) [reply tethysext/atcore/models/app_users/spatial_resource.py:84]: In other words I don't think lines 83 and 84 are necessary. Just use `self`

File: tethysext/atcore/tests/integrated_tests/models/app_users/spatial_resource_tests.py [hunk 2]
```diff
@@ -156,3 +157,12 @@ def test_get_extent_invalid_type(self):
             resource.get_extent('bad_type')
         self.assertTrue('"bad_type" is not a valid type for a SpatialResource '
                         'extent. Must be one of "wkt", "geojson", or "dict"' in str(exc.exception))
+
+    def test_update_extent_srid(self):
+        """Test trying to get an extent of an invalid type throws the correct error."""
+        resource = self.create_resource_in_session()
+        resource.extent = self.expected_geometry
+        resource.update_extent_srid(3857)
```
Comment (swainn) [inline tethysext/atcore/tests/integrated_tests/models/app_users/spatial_resource_tests.py:165]: Add an assert before you call `update_extent_srid` to verify that the SRID is not already 3857 (ie that it is actually changed).

**Review by swainn — CHANGES_REQUESTED** [review]
See comments.

File: tethysext/atcore/models/app_users/spatial_resource.py (hunk 1 above)
Comment (htran84) [reply tethysext/atcore/models/app_users/spatial_resource.py:84]: object_to_convert is resource. We need to do the query to get the geom extent

File: tethysext/atcore/tests/integrated_tests/models/app_users/spatial_resource_tests.py (hunk 2 above)
Comment (msouff) [reply tethysext/atcore/tests/integrated_tests/models/app_users/spatial_resource_tests.py:165]: We manually set it to 4326 (see line 40). There was no SRID before that.

Comment (htran84) [reply tethysext/atcore/tests/integrated_tests/models/app_users/spatial_resource_tests.py:165]: This is how we define the wkt. so if SRID=3857, it has changed.
self.extent_wkt = 'SRID=4326;POLYGON((-109.557466506958 40.4626541436629,-109.557123184204 40.4481880854746,' \
                          '-109.528369903564 40.4484166930046,-109.528756141663 40.4629153531036,-109.557466506958 ' \
                          '40.4626541436629))'

File: tethysext/atcore/models/app_users/spatial_resource.py (hunk 1 above)
Comment (htran84) [reply tethysext/atcore/models/app_users/spatial_resource.py:84]: got it.

Comment (msouff) [reply tethysext/atcore/models/app_users/spatial_resource.py:84]: Actually, that should work. We are working on it now.

**Review by swainn — APPROVED** [review]

**Pull request merged**